estargz/estargz.go

```diff
@@ -207,8 +207,15 @@ func (r *Reader) initFields() error {
 	uname := map[int]string{}
 	gname := map[int]string{}
 	var lastRegEnt *TOCEntry
-	for _, ent := range r.toc.Entries {
+	var chunkTopIndex int
+	for i, ent := range r.toc.Entries {
 		ent.Name = cleanEntryName(ent.Name)
+		if ent.Type == "reg" || ent.Type == "chunk" {
```

Comment:
better to use `switch{}`, but can be another PR

Reply:
Fixed to use `switch`.